prov/rxm/src/rxm_init.c

```diff
@@ -109,13 +109,23 @@ int rxm_alter_base_info(struct fi_info *base_info, struct fi_info **layer_info)
 	*info->ep_attr = *rxm_info.ep_attr;
 	*info->domain_attr = *rxm_info.domain_attr;
 
-	if (!(info->domain_attr->name =
-				ofi_strdup_add_prefix(base_info->domain_attr->name,
-					rxm_info.domain_attr->name))) {
```

Comment:
The assumption for the original code is the underlying provider is expected to fill in base_info->domain_attr->name. That's why the check for that was skipped. @shefty Is this a reasonable assumption to make?